### DISCUSSION TITLE
Is structuring a dandiset by experimental subject required?

### DISCUSSION THREAD

**Discussion by mo-osman** [dandi/helpdesk#72] [Q&A] [answered]
Hello,

I'm currently working with some folks from the Uchida Lab to upload the data from [this paper (Kim et al. 2020)](https://www.sciencedirect.com/science/article/pii/S0092867420315300) to DANDI. From what I've gathered, most dandisets are organized by experimental subject (i.e. the dandiset is a collection of directories containing the sessions associated with each subject). However, in our particular case, there are quite a few different session types (varying across recording modality, experimental manipulation, etc.) and some variation in which types of sessions each subject encountered. It therefore feels a bit more natural to structure the data by the type of session rather than subject identity (which is present in the NWB file if needed). What would you recommend?

Thanks,
Mo

**Comment by satra** [comment, marked as answer]
@mo-osman - for the moment i would keep the organization the same. since session is included in the NWB file and hence in the filename, one would be able to filter by session if wanting to restrict the search. 

the primary reason for using subject as a more primary  identifier than a session is to be in alignment with the BIDS standard.

**Reply by mo-osman** [reply]
Thanks for the quick reply. In that case, I'll structure by subject. Cheers

**Reply by satra** [reply]
btw, just a note that the `dandi organize` step when done on NWB files will automatically do this for you. on your side it's important to ensure that you have both a subject id and a session id in the nwb file. the nwb inspector developed by @bendichter can help determine if your nwb files are appropriately constructed.
